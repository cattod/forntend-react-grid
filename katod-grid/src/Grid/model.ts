export interface ICatodcolumnDefs<T> {
  title:string,
   icon?: string,
   key: string,
    valueGetter?(data:T):any,
     displayValue? ():void
}


export interface IAgColumnDefs<T> {
    headerName ?: string,
     field ?: string
     cellRenderer?:string,
     width ?: number
     headerComponent ?: any
     headerComponentParams ?:any
     suppressMenu?:boolean
     frameworkComponents?: any
     cellEditorFramework? : ICatodActions<T>[]|undefined
     sortable?: boolean
     pinned?: string
     suppressSizeToFit?:boolean
     minWidth?: number,
     maxWidth?:number
}


export interface ICatodActions<T> {
  title:string,
  key?: string,
  icon :string,
  actionFn ?(data?: T):void,

}

export interface IMessage{
  emptyData?:string
  loading?:string
}