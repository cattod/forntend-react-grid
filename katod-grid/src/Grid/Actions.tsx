import React , {Component} from "react"
import {IActions,ICatodcolumnDefs} from "./model"

interface IProps<T>{
  //  id ?:Array <columnDefs>  
  colDef : ICatodcolumnDefs<T>
  data: T
}

interface IState<T>{
    actions: IActions<T>[]
}

export class Actions<T> extends Component<IProps<T>,IState<T>>{
    constructor(props:IProps<T>){
        super(props)
        this.state = {actions:[]}
    }

    componentDidMount() {
        // if (this.props.colDef.cellEditorFramework) {
        //     this.setState({actions:this.props.colDef.cellEditorFramework})
        // }
    }

    render() {
       
        return(
            <React.Fragment>
                {this.state.actions? this.state.actions.map((item: IActions<T>,index:number)=>{
                return <button key={index} onClick={()=>{
                    if (item.actionFa)
                    item.actionFa(this.props.data)
                   
                }}>{item.title}</button>
                }):null}
               
            </React.Fragment>
        )
    }

}