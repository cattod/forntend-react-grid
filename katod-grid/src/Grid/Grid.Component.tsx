import React , {Component} from "react";
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';
import { AgGridReact } from 'ag-grid-react';
import {columnDefs} from "./model"
import {defineCol} from "./utils"

interface IState<T>{
    columnDefs:Array <columnDefs> 
    rowData: T[]
}

interface IProps<T,P>{
    columnDefs?:Array <columnDefs>  
    rowData?:T[]
    frameworkComponents:P
}

export class CatodGrid<T,P> extends Component<IProps<T,P>,IState<T>> {
    constructor(props:IProps<T,P>) {
        super(props);
        this.state = {
          columnDefs: [],
          rowData: []
        }
      }


      componentDidMount() {
        if (this.props.columnDefs) {
          this.setState({columnDefs:this.props.columnDefs})
        } else{
          this.setState({columnDefs: defineCol(this.props.rowData)})
        }
        
      }


    render() {
        return(
            <div className="ag-theme-balham" style={ {height: '200px', width: '600px'} }>
            <AgGridReact
                columnDefs={this.state.columnDefs}
               rowData={this.props.rowData}
              frameworkComponents={this.props.frameworkComponents}
              >
            </AgGridReact>
          </div>
        )
    }

}






