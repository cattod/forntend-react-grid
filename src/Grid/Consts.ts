export enum EnumConsts {
    ThereIsNotAnyDataToShowInGrid="There is not any data to show in grid",
    None = "none",
    RTL ="rtl",
    Action = "action",
    Descending="descending",
    Ascending="ascending",
    Right = "right",
    Left = "left"


}