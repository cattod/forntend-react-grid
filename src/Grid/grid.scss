
$background-color-prime:#f6f6f6;
$border-color :  #ddd;
$text-color: rgb(68, 68, 68);
$header-text-color : #444;


.action-padding {

    margin: 1px;
    span{
        padding: 2px;
    }
}

tr:nth-child(odd){
    background-color:$background-color-prime;
 }

 .empty-grid{
     text-align: center;
     border: 1px solid $border-color;
     width:100%;
  margin:5px auto;
  padding: 6px;
  border-radius: 5px;

 }

 .empty-dataRow{ 
     text-align: center;
        }

 .grid-body{
     margin: 3px;
     font-size: 13px;   
    
     table{
         color: $text-color;        
       thead{
        color:$header-text-color;       
        th{
            font-weight: 600;           
        }
       }
     }
 }

 .grid-body-right {
     direction: rtl;
    margin: 3px;
    font-size: 13px;
   text-align: right;   
    table{
        color: $text-color;       
      thead{
       color:$header-text-color;      
       th{
           font-weight: 600;          
       }
      }
    }
 }


      .action-cattod{
   display:none;
  
        button{
            font-size: 13px;
            color:$header-text-color;

        }
        
          }

      .row-hover-catod tbody tr:hover{
         
          .action-cattod{
           
            display: inline;
            button{
                font-size: 13px;
                color:$header-text-color;
               
                
    
            }
          }
      }

   .sort-icon{
    width:12px;
     margin-right:3px;
   }

   .sort-svg{
    fill:none;
    stroke:$header-text-color;
    stroke-width:1
   }

   .scroll
{
overflow-x:auto;
}

.break-text-grid{
    word-break: break-all;
}

.image-icon{
    width: 10px;
}