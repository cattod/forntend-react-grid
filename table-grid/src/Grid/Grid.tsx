import React, { Component } from "react"
import { ICatodActions, ICatodcolumnDefs } from "./Model"
import { Actions } from "./Actions"

interface IProps<T> {
    columnDef?: Array<ICatodcolumnDefs<T>>
    dataRow?: T[]
    actions?: ICatodActions<T>[]
    rowNumber?: number
    height?: string
    message?: string
}

interface IState<T> {
    dataRow?: T[]
    loading: boolean,
    headerDef: Array<ICatodcolumnDefs<T>> | undefined
    message: string,
    textAlign: boolean,
    sortType: "ascending"|"descending"|"none"
}

export class Grid<T> extends Component<IProps<T>, IState<T>> {
    _textAlign :boolean = true
    constructor(props: IProps<T>) {
        super(props)
        this.state = { loading: false,
             headerDef: undefined,
              message: "There is not any data for show in grid",
              textAlign:true,
              sortType:"none",
              dataRow:[]
             }


    }

    componentDidUpdate(prevProps: IProps<T>) {
    
    
        if (this.props.dataRow !== prevProps.dataRow) {
        
            this.setState({  dataRow:this.props.dataRow})
    
        }
      }

    componentDidMount() {
        const element: Element | null = document.querySelector('.grid-body')
        const style: CSSStyleDeclaration | "" = element ? getComputedStyle(element) : ""
        let _TA = this.state.textAlign
        if (style !== "") {
         
          if (style.direction === "rtl") {
            _TA = false
          }
        }
        let newHeader: Array<ICatodcolumnDefs<T>> | undefined = undefined
        let newDataRow:T[] = []
        if (this.props.dataRow) {
            newDataRow = this.props.dataRow
        }
        if (this.props.dataRow && !this.props.columnDef) {
            newHeader = this.createHeader(this.props.dataRow)
            
        } else if (this.props.columnDef) {
            newHeader = this.props.columnDef
            // this.setState({headerDef:this.props.columnDef})
        }

        if (this.props.actions) {
            newHeader?.push({ title: "", key: "action", icon: "" ,displayValue:()=>{return""}, valueGetter:()=>{return ""}})
        }
        this.setState({ headerDef: newHeader, textAlign:_TA , dataRow:newDataRow})
    }

    createHeader = (newData: T[]): Array<ICatodcolumnDefs<T>> => {
        //    let newData =  this.props.dataRow 

        // let newCol = Object.keys(newData[0])  
        let newCol = Object.keys(newData[0]).map((item) => {
            return {
                title: item[0].toUpperCase() + item.slice(1),
                key: item,
                icon: "",
                displayValue:()=>{return""},
                valueGetter:()=>{return""}
            }

        })

        return newCol



    }

    tableRow<T, K extends keyof T>(rowData: T, colDef: ICatodcolumnDefs<T>, key: K, actions?: ICatodActions<T>[]) {


        if (key === "action" && actions) {
            return <Actions rowData={rowData} colDef={colDef} actionsCatod={actions} />
        } else {
            return colDef.displayValue(rowData)
//             if (colDef.displayValue && colDef.valueGetter) {

//                 return rowData[key]
//             } else if (colDef.displayValue) {

//                 return colDef.displayValue(rowData)
//             }

//             else if (colDef.valueGetter) {
//                 return colDef.valueGetter(rowData)
//             } else return rowData[key]
        }
    }
    sortHandle = (sortType:"ascending"|"descending"|"none", onSort:any)=>{
      
        this.setState({sortType:sortType, dataRow:onSort(sortType)})
    }

    headerTemplate = (data:ICatodcolumnDefs<T>)=>{
        if (data.onSort){
            switch(this.state.sortType){
                case "ascending":                 
                return <div >{data.title}
                <img  onClick={()=>this.sortHandle("descending",data.onSort)} className="sort-icon"  src="dascending-sort.svg"/> </div>
                case "descending": 
                return <div >{ data.title}
                <img  onClick={()=>this.sortHandle("none",data.onSort)} className="sort-icon" src="ascending-sort.svg"/>
                </div>
                case "none":
                    return <div >{ data.title}
                    <img  onClick={()=>this.sortHandle("ascending",data.onSort)} className="sort-icon" src="sort.svg"/>
                    </div>
                default: return
            }
          
        }
    
return data.title
    }

    render() {

        return (

            <div className={this.state.textAlign?"grid-body":"grid-body-right"} >
                {this.state.loading ? <div>loading...</div> :
                    <table className="table table-bordered table-sm table-hover">
                        <thead className="back-header">
                            <tr>
                                {this.state.headerDef?.map((item, index) => {
                                    return (
                                        <th scope="col" key={item.key}>{this.headerTemplate(item)}</th>
                                    )
                                })}

                            </tr>
                        </thead>
                        <tbody>
                            {this.state.dataRow?.map((item: any, index:number) => {

                                return (
                                    <tr key={index}>
                                        {this.state.headerDef?.map((element: any, id) => {
                                            return (
                                                <td key={id}>
                                                    {this.tableRow(item, element, element.key, this.props.actions)}

                                                </td>

                                            )
                                        })}
                                    </tr>
                                )
                            })}

                        </tbody>
                    </table>}

                {!this.props.columnDef && !this.props.dataRow ? <p className="empty-grid">{this.props.message ? this.props.message : this.state.message}</p> : null}
            </div>

        )
    }

}