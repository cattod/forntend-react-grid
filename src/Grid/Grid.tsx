import React, { Component } from "react"
import { ICatodActions, ICatodcolumnDefs, ISort, ICattodGridProps,IDefaultSort } from "./Model"
import { Actions } from "./Actions"
import { EnumConsts } from "./Consts"

// interface type for props of GridCattod
interface IProps<T> extends ICattodGridProps<T> { }

//state interface type
interface IState<T> {
    headerDef: Array<ICatodcolumnDefs<T>> | undefined
    message: string,
    textAlign: boolean,
    sortType:IDefaultSort[]
    lastSortType:IDefaultSort[]
}

//CatodGrid component
 class GridClass<T> extends Component<IProps<T>, IState<T>> {

    constructor(props: IProps<T>) {
        super(props)
        this.state = {
            lastSortType:[],
            headerDef: undefined,
            message: EnumConsts.ThereIsNotAnyDataToShowInGrid,
            textAlign: true,
            sortType: []
            // sortType:{sortType:EnumConsts.None}     
        }
    }

    //prepare data for rendering
     /**
   * prepare component after mounting componet.
   * fix direction of the component for stylesheet
   * prepare actions column if there is actions in props
   *
   * @remarks
   * This method is part of the {@link core-library#React | React subsystem}.
   *
   *
   * @beta
   */

    componentDidMount() {
        const element: Element | null = document.querySelector('.grid-body')
        const style: CSSStyleDeclaration | "" = element ? getComputedStyle(element) : ""
        let _TA:boolean = this.state.textAlign
        let newHeader: Array<ICatodcolumnDefs<T>> = this.props.columnDef
        let newSortType: IDefaultSort[] = []

        //check for direction in props and parent project direction
        if (this.props.direction) {
            _TA = this.props.direction === "right" ? false : true
        } else {
            if (style !== "") {

                if (style.direction === EnumConsts.RTL) {
                    _TA = false
                }
            }
        }



        //check for action in grid props and add column to grid
        if (this.props.actions) {
            newHeader?.push({
                title: "",
                key: EnumConsts.Action,
                
                displayValue: () => { return "" },
               
            })
        }
        if(newHeader.length){
            for (let item of newHeader){
               if (item.sortable){
                  
                newSortType.push({columnKey:item.key, sortType:"none"})
                
               } 
            }
          
        }
        
        if (this.props.defaultSort?.length){
            for (let j in this.props.defaultSort) {
                for (let i in newSortType){
                    if (this.props.defaultSort[j].columnKey === newSortType[i].columnKey){
                       
                       newSortType.splice(Number(i), 1)                       
                        newSortType.unshift(this.props.defaultSort[j])
                    }
             
                 }
            }
          
        }
        //update state for new changes
        this.setState({ headerDef: newHeader, textAlign: _TA , sortType:newSortType,lastSortType:newSortType})
    }

    /**
   * Returns cell value to display
.
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @param rowData - The first input <T>
   * @param colDef - The second input ICatodcolumnDefs<T>
   * @param key - The third input string
   * @param actions - The forth input ICatodActions<T>[]
   * @returns - The displayable value
   *
   * @beta
   */

    rowCell=(rowData: T, colDef: ICatodcolumnDefs<T>, key: string, actions?: ICatodActions<T>[]):
    React.FunctionComponent | React.ReactNode|string|number|boolean=> {
        // there is a discrete for action data, which need our component definition
        if (key === EnumConsts.Action && actions) {
            return <Actions
                rowData={rowData}
                colDef={colDef}
                actionsCatod={actions}
            />
        } else {
            return colDef.displayValue(rowData)
        }
    }

     /**
   * Run onSort function in props and update sort type for column.
   *
   * @remarks
   * This method is called when sort icon clicked.
   *
   * @param sortType - The first input ISort
   * @param key - The second input string
   *
   * @beta
   */

   async sortHandle (sortType: ISort, key: string) {
        let newSortType = this.state.sortType
        
        if (!this.props.multiSort){
            for (let j in this.state.sortType) {  
                newSortType[j].sortType ="none"         
                if (this.state.sortType[j].columnKey ===key){
                   
                   newSortType.splice(Number(j), 1)                       
                    newSortType.unshift({sortType:sortType.sortType, columnKey:key})
                }
         
            
        }
        } else {
            for (let j in this.state.sortType) {           
                if (this.state.sortType[j].columnKey ===key){
                   
                   newSortType.splice(Number(j), 1)                       
                    newSortType.unshift({sortType:sortType.sortType, columnKey:key})
                }
         
            
        }
        }
          
        if (this.state.headerDef) {
            console.log(this.state.headerDef)

            if (this.props.onSort) {
              await  this.props.onSort(newSortType)               
                this.setState({ sortType: newSortType })
               
            }
           
        }
    }



     /**
   * Returns header value to display.
   * Add sort icon to header of table for sortable column and set header title
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @param data - The first input  ICatodcolumnDefs<T>
   * @returns - The displayable value in the header of every column
   *
   * @beta
   */
    selectSortType = (data: ICatodcolumnDefs<T>) => {
        
        if (data.sortable) {
            for (let i in this.state.sortType) {
                if (data.key === this.state.sortType[i]?.columnKey) {
                    switch (this.state.sortType[i].sortType) {
                        case EnumConsts.Ascending:
                            return <div >{data.title}{this.upSvg(data)}</div>
                        case EnumConsts.Descending:
                            return <div >{data.title}{this.downSvg(data)}</div>
                        case EnumConsts.None:
                            return <div >{data.title}{this.unsortSVG(data)}</div>
                        default: return <div >{data.title}{this.unsortSVG(data)}</div>
                    }
                } else {
                    if (!this.props.multiSort) {
                        return <div >{data.title}{this.unsortSVG(data)}</div>
                    }

                }
               
            }
           

       }


        return data.title
    }

         /**
   * Returns svg for unsort icon.
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @param data - The first input  ICatodcolumnDefs<T>
   * @returns - The svg for unsort icon.
   *
   * @beta
   */
    unsortSVG = (data: ICatodcolumnDefs<T>) => {
        return <svg height="15" width="20" name="sort"
            onClick={() => this.sortHandle({ sortType: EnumConsts.Ascending }, data.key)}
        >
            <polyline points="8,5 11,0 14,5 11,0 11,15"
                className="sort-svg"
            />
            <polyline points="5,0 5,15 2,10 5,15 8,10" className="sort-svg" />
     Sorry, your browser does not support inline SVG.
   </svg>
    }

            /**
   * Returns svg for descending sort icon.
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @param data - The first input  ICatodcolumnDefs<T>
   * @returns - The svg for descending sort icon.
   *
   * @beta
   */
    upSvg = (data: ICatodcolumnDefs<T>) => {
        return <svg height="15" width="10" name="up"
            onClick={() => this.sortHandle({ sortType: EnumConsts.Descending }, data.key)}
        >
            <polyline points="2,5 5,0 8,5 5,0 5,15" className="sort-svg" />
        Sorry, your browser does not support inline SVG.
      </svg>
    }
    
            /**
   * Returns svg for ascending sort icon.
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @param data - The first input  ICatodcolumnDefs<T>
   * @returns - The svg for ascending sort icon.
   *
   * @beta
   */
    downSvg = (data: ICatodcolumnDefs<T>) => {
        return <svg height="15" width="10" name="up"
            onClick={() => this.sortHandle({ sortType: EnumConsts.None }, data.key)}
        >
            <polyline points="5,0 5,15 2,10 5,15 8,10 " className="sort-svg" />
        Sorry, your browser does not support inline SVG.
      </svg>
    }

                /**
   * Returns the displayable xhtml for different condition.
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @returns - The displayable xhtml for different condition.
   *
   * @beta
   */
    displayGrid = () => {

        if (this.props.dataRow.length < 1 && this.props.columnDef.length < 1) {
            return <p className="empty-grid">
                {this.props.message ?
                    this.props.message :
                    this.state.message}
            </p>

        }
        else if (this.props.dataRow.length < 1 && this.props.columnDef.length > 0) {
            return (
                <div>
                    {this.tableGrid()}
                    <p className="empty-grid">{this.props.message ? this.props.message : this.state.message}</p>
                </div>
            )

        }
        return this.tableGrid()

    }
                    /**
   * Returns the displayable data table.
   *
   * @remarks
   * This method is part of the {@link core-library#Grid | Grid subsystem}.
   *
   * @returns - The displayable data table.
   *
   * @beta
   */
    
    tableGrid = () => {
        //
        return (
            <table 
            className={`${this.props.className} table row-hover-catod  table-hover`}>
           
                <thead className="back-header">
                    <tr>
                        {this.state.headerDef?.map((item: ICatodcolumnDefs<T>) => {
                            return (
                                <th scope="col" key={item.key} id={item.key}>{this.selectSortType(item)}</th>
                            )
                        })}

                    </tr>
                </thead>
                <tbody>
                    {this.props.dataRow.map((item: T, index: number) => {

                        return (
                            <tr key={index} >
                                {this.state.headerDef?.map((element: ICatodcolumnDefs<T>, id: number) => {
                                    return (
                                        <td key={id} >
                                            {this.rowCell(item, element, element.key, this.props.actions)}

                                        </td>

                                    )
                                })}
                            </tr>
                        )
                    })}

                </tbody>
            </table>
        )
    }

    render() {

        return (

            <div className={this.state.textAlign ? "grid-body" : "grid-body-right"}
            >
                {this.displayGrid()}

            </div>

        )
    }

}

export const Grid = React.memo(GridClass);

