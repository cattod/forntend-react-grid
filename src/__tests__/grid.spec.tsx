import * as React from "react";
import {CatodGrid, Catodcolumn} from "../index";
import {create} from "react-test-renderer";
import {EnumConsts} from "../Grid/Consts"
// let i = 0;
// afterEach(() => {
//   i++;
//   return console.log(i);
// });

// afterEach(cleanup)

test("Component should show 'red' text 'Hello World'", () => {
  const component = create(<CatodGrid columnDef={[]} dataRow={[]} message="Hello World"/>);
  const testInstance = component.root;

  expect(testInstance.findByType(CatodGrid).props.message).toBe("Hello World");

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test(`Component should show ${EnumConsts.ThereIsNotAnyDataToShowInGrid}`, () => {
  const component = create(<CatodGrid columnDef={[]} dataRow={[]} />);
  const testInstance = component.root;

  expect(testInstance.findByProps({className: "empty-grid"}).children).toEqual([EnumConsts.ThereIsNotAnyDataToShowInGrid]);

  // let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});

test(`Component should show the table whith 2 column which the titles are "Make" and "Model" and message ${EnumConsts.ThereIsNotAnyDataToShowInGrid}`, () => {
  interface IRowData{
    make ?:string,
    model ?: string,
    price ?: number,
    }
  let columns:Catodcolumn<IRowData>[] = [{title:"Make",
      
  key:"Make",
  displayValue: (rowData:IRowData) =>{return rowData?.make}
  },
  {title:"Model",
 
  key:"Model",
  displayValue:(rowData:IRowData)=>{return rowData?.model},
  valueGetter:(rowData:IRowData)=>{return rowData?.model}
  },]
  const component = create(<CatodGrid columnDef={columns} dataRow={[]} />);
  const testInstance = component.root;
  let dataZero = testInstance.findByType(CatodGrid).props.dataRow[0]
  expect(testInstance.findByType(CatodGrid).props.columnDef[0].key).toBe("Make");
  expect(testInstance.findByType(CatodGrid).props.columnDef[1].key).toBe("Model");
  expect(testInstance.findByType(CatodGrid).props.columnDef[0].displayValue(dataZero)).toBeUndefined();
  expect(testInstance.findByProps({className: "empty-grid"}).children).toEqual([EnumConsts.ThereIsNotAnyDataToShowInGrid]);
 
  // let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});

test(`Component should show the table whith 2 column which the titles are "Make" and "Model" whith sort icon`, () => {
  interface IRowData{
    make ?:string,
    model ?: string,
    price ?: number,
    }
  let columns:Catodcolumn<IRowData>[] = [{title:"Make",
      sortable:true,
  key:"Make",
  displayValue: (rowData:IRowData) =>{return rowData?.make}
  
  },
  {title:"Model",
 
  key:"Model",
  displayValue:(rowData:IRowData)=>{return rowData?.model},
  valueGetter:(rowData:IRowData)=>{return rowData?.model}
  },]
  const component = create(<CatodGrid columnDef={columns} dataRow={[]} />);
  const testInstance = component.root;
  let dataZero = testInstance.findByType(CatodGrid).props.dataRow[0]
  expect(testInstance.findByType(CatodGrid).props.columnDef[0].key).toBe("Make");
  expect(testInstance.findByType(CatodGrid).props.columnDef[1].key).toBe("Model");
  // expect(testInstance.findByProps({className:"sort-svg"}).children).toBeCalled();
  expect(testInstance.findByType(CatodGrid).props.columnDef[0].displayValue(dataZero)).toBeUndefined();
  expect(testInstance.findByProps({className: "empty-grid"}).children).toEqual([EnumConsts.ThereIsNotAnyDataToShowInGrid]);
 
  // let tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});