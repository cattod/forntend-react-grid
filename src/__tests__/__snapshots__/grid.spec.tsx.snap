// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should show 'red' text 'Hello World' 1`] = `
<div
  className="grid-body"
>
  <p
    className="empty-grid"
  >
    Hello World
  </p>
</div>
`;
